quill-cassandra-ce/src/main/scala-2.12/io/getquill/util/GuavaCeUtils.scala

```diff
@@ -0,0 +1,28 @@
+package io.getquill.util
+
+import com.google.common.util.concurrent.{ FutureCallback, Futures, ListenableFuture }
+
+import cats.effect.Async
+import scala.concurrent.ExecutionContext.global
+
+import java.util.concurrent.Executor
+import scala.language.higherKinds
+
+object GuavaCeUtils {
+  implicit class RichListenableFuture[T](lf: ListenableFuture[T]) {
+
+    case object ecExecutor extends Executor {
+      def execute(command: Runnable): Unit = global.execute(command)
+    }
+
+    def toAsync[F[_]: Async]: F[T] = {
```

Comment:
So in general why do we need `toAsync` that converts `F[_]: Async` to a `F[T]` ? I am not that familiar with CE but I would like some context on this. Also if this is really needed I don't think putting this code in a top level `object` is appropriate, it makes more sense to put it inside one of the DSL traits that automatically included

Reply:
`toAsync` converts `ListenableFuture[T]` to `F[T]`, where `F` dependant on `Async`.

ZIO did similar thing in ZioConversions object.

Reply:
I also think this is also reasonable.

Reply:
This is fine. The Cassandra session uses Guava futures directly so you have to wrap them into `F[_]`. You either do that or re-implement Cassandra's entire effect stack which would be an insane effort.